linked_list.rb

```diff
@@ -42,17 +42,41 @@ def search(value)
   # method to return the max value in the linked list
   # returns the data value and not the node
   def find_max
-    puts "Not implemented"
+    current = @head
+    max = current.data
+    while current.next
+      current = current.next
+      if current.data > max
+        max = current.data
+      end
+    end
+    return max
   end
 
   # method to return the min value in the linked list
   # returns the data value and not the node
   def find_min
+    current = @head
+    min = current.data

```

Comment:
Same as with find_max, check for null before accessing member variables in the node.